chromoscores/snipping.py

```diff
@@ -1,47 +1,88 @@
 import numpy as np
 
 
-def peak_snipping(contact_map, window_size, peak_coordinate):
-    """
-    Peak snippet; snippet with a "window_size" around peak with "peak_coordinate"
-    on a contact map, "contact_map". peak_coordinate should be in the format
-     of (i,j)
-    -----------------
-    Function peak_snippet(contact_map, window_size, peak_coordinate):
-
-    begin function
-
-         raise an error if peak coordinate + window window_size is out of the map
-
-         snippet = contact_map[
-        (peak_coordinate[0] - window_size):(peak_coordinate[0] + window_size),
-        (peak_coordinate[1] - window_size):(peak_coordinate[1] + window_size)]
 
-     return snippet_matrix
+def get_diagonal_pile_up(contact_map, boundary_list, window_size):
+    """
+    parameters
+    ----------
+    contact_map: contact map
+    boundary_list: list of the boundary elements positions on the diagonal 
+    window_size: size of the window
+
+    Returns
+    -------
+    a stackup of snippts around the boundary elements
+    """
+    mat = np.zeros((window_size, window_size))
+    for i in range(len(boundary_list)):
+        mat += contact_map[
+            boundary_list[i] - window_size // 2 : boundary_list[i] + window_size // 2,
+            boundary_list[i] - window_size // 2 : boundary_list[i] + window_size // 2,
+        ]
+    return mat
+
+def get_expected_map(contact_map):
+    """
+    parameters
+    ----------
+    contact_map: contact map
 
-     end function
-    ----------------
+    Returns
+    -------
+    a normalized contact map based on the average of each diagonal from the main diagonal
     """
+    mat = np.zeros(np.shape(contact_map))
+    for i in range(len(contact_map)):
+        for j in range(len(contact_map) - i):
+            mat[i, i + j] = contact_map[i, i + j] / (
+                np.mean(np.diag(contact_map, k=j))
+            )
+            mat[i + j, i] = mat[i, i + j]
+    return mat
 
+def peak_snipping(contact_map, window_size, peak_coordinate):
+    """
+    parameters
+    ----------
+    contact_map: contact map
+    window_size: size of the window
+    peak_coordinate: coordinate of the peak in (i,j) format
+
+    returns
+    -------
+    a snippet of the contact map around the specified peak
+    """
     if (peak_coordinate[1] + window_size) > len(contact_map) or (
         peak_coordinate[0] - window_size
     ) < 0:
-
         raise ValueError(
             "window window_size for peak coordinate exceeds window_size of the contact map"
         )
-
+    
     snippet = contact_map[
         (peak_coordinate[0] - window_size) : (peak_coordinate[0] + window_size),
         (peak_coordinate[1] - window_size) : (peak_coordinate[1] + window_size),
     ]
-
     return snippet
 
 
-def get_isolation_snippets(
-    contact_map, delta=1, diag_offset=1, max_distance=10, state=1
+def get_isolation_areas(
```

Comment:
this can be moved to the scorefunctions and made a helper function (and then the import wouldn't be needed either!). Note the convention for distinguishing helper functions is to put an underscore before the name: e.g. _ get_isolation_areas